test/functional/feature_reindex_readonly.py

```diff
@@ -50,15 +50,30 @@ def reindex_readonly(self):
                     self.log.warning("Return early on Linux under root, because chattr failed.")
                     self.log.warning("This should only happen due to missing capabilities in a container.")
                     self.log.warning("Make sure to --cap-add LINUX_IMMUTABLE if you want to run this test.")
-                    return
-
-        self.log.debug("Attempt to restart and reindex the node with the unwritable block file")
-        with self.nodes[0].assert_debug_log(expected_msgs=['FlushStateToDisk', 'failed to open file'], unexpected_msgs=[]):
-            self.nodes[0].assert_start_raises_init_error(extra_args=['-reindex', '-fastprune'],
-                expected_msg="Error: A fatal internal error occurred, see debug.log for details")
+                    undo_immutable = False
+        if platform.system() == "FreeBSD" or platform.system() == "OpenBSD":
+            try:
+                subprocess.run(['chflags', 'uchg', filename], capture_output=True, check=True)
+                undo_immutable = subprocess.check_call(['chflags', 'nouchg', filename])
+                self.log.info("Made file immutable with chflags")
+            except subprocess.CalledProcessError as e:
+                self.log.warning(str(e))
+                if e.stdout:
+                    self.log.warning(f"stdout: {e.stdout}")
+                if e.stderr:
+                    self.log.warning(f"stderr: {e.stderr}")
+                if os.getuid() == 0:
+                    self.log.warning("Return early on BSD under root, because chflags failed.")
+                    undo_immutable = False
 
-        if used_chattr:
-            subprocess.check_call(['chattr', '-i', filename])
+        if undo_immutable:
+            self.log.debug("Attempt to restart and reindex the node with the unwritable block file")
+            with self.nodes[0].assert_debug_log(expected_msgs=['FlushStateToDisk', 'failed to open file'], unexpected_msgs=[]):
+                self.nodes[0].assert_start_raises_init_error(extra_args=['-reindex', '-fastprune'],
+                    expected_msg="Error: A fatal internal error occurred, see debug.log for details")
+            undo_immutable()
+        else:
+            return
```

Comment:
heh thanks. Having an issue on macos now...

Reply:
@jonatack as root?

Reply:
OK I see even before this new commit macOS test would work as root. Hm I thought we already went over that. I guess the macOS ci is not running as root?

Reply:
yes, macos as root on master doesnt work. However, this will:

`if platform.system() == "FreeBSD" or platform.system() == "OpenBSD" or platform.system() == "Darwin":`

whats the best style for a triple - or in python?

Reply:
`if platform.system() in ["FreeBSD", "OpenBSD", "Darwin"]:` there we go ;-)

Reply:
@maflcko the first commit does not affect macos. the second commit is a refactor that also fixes a macos/root bug on master. Want me to separate the macos out to commit 3?